subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/tasks/GeneratePluginDescriptors.java

```diff
@@ -41,15 +45,20 @@
 public class GeneratePluginDescriptors extends DefaultTask {
 
     private final ListProperty<PluginDeclaration> declarations;
+    private final Provider<Map<String, String>> implementationClassById;
     private final DirectoryProperty outputDirectory;
 
     public GeneratePluginDescriptors() {
         ObjectFactory objectFactory = getProject().getObjects();
         declarations = objectFactory.listProperty(PluginDeclaration.class);
         outputDirectory = objectFactory.directoryProperty();
+        implementationClassById = getDeclarations().map(declarations -> declarations.stream()
+            .collect(Collectors.toMap(PluginDeclaration::getId, PluginDeclaration::getImplementationClass, (a, b) -> b, LinkedHashMap::new))
+        );
+        getInputs().property("implementationClassById", implementationClassById);
```

Comment:
Why expose this via the runtime API and not via a getter?

Reply:
Exposing this via a getter is a little bit strange:
- I need to declare it in the acceptedAPI changes, although I declared it `protected`.
- Other tasks/plugins could start using it, and this is purely an internal thing.

So I opted for using the runtime API, since this hides the property from the API of the task. Any objections?

Reply:
Would using package private instead of `protected` solve these problems?

Reply:
I change this to use package private.